public/app/features/transformers/editors/FilterByNameTransformerEditor.tsx

```diff
@@ -161,19 +177,46 @@ export class FilterByNameTransformerEditor extends React.PureComponent<
     this.setState({ isRegexValid });
   };
 
+  onVariableChange = (selected: SelectableValue) => {
+    this.props.onChange({
+      ...this.props.options,
+      include: { variable: selected.value },
+    });
+
+    this.setState({ variable: selected.value });
+  };
+
+  onFromVariableChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const val = e.currentTarget.checked;
+    this.props.onChange({ ...this.props.options, byVariable: val });
+    this.setState({ byVariable: val });
+  };
+
   render() {
     const { options, selected, isRegexValid } = this.state;
     return (
-      <div className="gf-form-inline">
-        <div className="gf-form gf-form--grow">
-          <div className="gf-form-label width-8">Identifier</div>
-          <HorizontalGroup spacing="xs" align="flex-start" wrap>
-            <Field
+      <div>
+        <InlineFieldRow label="Use variable">
+          <InlineField label="From variable">
+            <InlineSwitch value={this.state.byVariable} onChange={this.onFromVariableChange}></InlineSwitch>
```

Comment:
React is throwing a `Warning: A component is changing an uncontrolled input to be controlled` when setting this transformation and switching to `From variable`. I think we should fix that.